core/src/main/scala/chisel3/Module.scala

```diff
@@ -347,12 +347,16 @@ package experimental {
 
     // These methods allow checking some properties of ports before the module is closed,
     // mainly for compatibility purposes.
-    protected def portsContains(elem: Data): Boolean = _ports contains elem
+    protected def portsContains(elem: Data): Boolean = {
+      _ports.exists { port => port._1 == elem }
+    }
 
     // This is dangerous because it can be called before the module is closed and thus there could
     // be more ports and names have not yet been finalized.
     // This should only to be used during the process of closing when it is safe to do so.
-    private[chisel3] def findPort(name: String): Option[Data] = _ports.find(_.seedOpt.contains(name))
+    private[chisel3] def findPort(name: String): Option[Data] = {
+      _ports.map(_._1).find { _.seedOpt.contains(name) }
```

Comment:
```suggestion
      _ports.find(_._1.seedOpt.contains(name))
```
Unless the JIT can optimize, that `.map` is going to create a copy of `_ports` every time this function is called which would be pretty expensive.

Reply:
Makes sense. will fix on next push